pkg/operator/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,292 @@
+		CurrentHostnames: []configv1.Hostname{configv1.Hostname(route.Spec.Host)},
+		ConsumingUsers:   c.consumingUsers,
+		RelatedObjects: []configv1.ObjectReference{
+			{
+				Group:     routev1.GroupName,
+				Resource:  "routes",
+				Name:      route.ObjectMeta.Name,
+				Namespace: route.ObjectMeta.Namespace,
+			},
+		},
+	}
+	conditions := checkErrorsConfiguringCustomRoute(customRouteErrors)
+	if conditions == nil {
+		conditions = checkIngressURI(ingressConfig, route)
+		if conditions == nil {
+			conditions = checkRouteAvailablity(c.secretLister, ingressConfig, route)
+		}
+	}
+	componentRoute.Conditions = ensureDefaultConditions(conditions)
+	_, err := c.updateComponentRouteStatus(ctx, componentRoute)
+	return err
+}
+
+func (c *customRouteController) syncSecret(secretName string) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	if secretName != "" {
+		source = resourcesynccontroller.ResourceLocation{Namespace: "openshift-config", Name: secretName}
+	}
+
+	destination := resourcesynccontroller.ResourceLocation{Namespace: c.destSecret.Namespace, Name: c.destSecret.Name}
+
+	return c.resourceSyncer.SyncSecret(destination, source)
+}
+
+// updateComponentRouteStatus searches the entries of the ingress.status.componentRoutes array for a componentRoute with a matching namespace and name.
+// If a matching componentRoute is found, the two objects are compared minus any of the conditions.lastTransactionTime entries. If all the fields
+// match, the entry is updated.
+// If no matching componentRoute is found, the entry is appended to the list.
+// true is returned if the status of the ingress.config.openshift.io/cluster is updated.
+func (c *customRouteController) updateComponentRouteStatus(ctx context.Context, componentRoute *configv1.ComponentRouteStatus) (bool, error) {
+	// Override the timestamps
+	now := metav1.Now()
+
+	// Create a copy for comparison and set transaction times
+	componentRouteCopy := componentRoute.DeepCopy()
+	setLastTransactionTime(componentRouteCopy, now)
+
+	updated := false
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		ingressConfig, err := c.ingressClient.Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		existingComponentRoute := routecomponenthelpers.GetComponentRouteStatus(ingressConfig, componentRoute.Namespace, componentRoute.Name)
+		if existingComponentRoute != nil {
+			// Create a copy for comparison and remove transaction times
+			existingComponentRouteCopy := existingComponentRoute.DeepCopy()
+			setLastTransactionTime(existingComponentRouteCopy, now)
+
+			// Check if an update is needed
+			if equality.Semantic.DeepEqual(componentRouteCopy, existingComponentRouteCopy) {
+				return nil
+			}
+			*existingComponentRoute = *componentRoute
+		} else {
+			ingressConfig.Status.ComponentRoutes = append(ingressConfig.Status.ComponentRoutes, *componentRoute)
```

Comment:
the last transition time is very likely missing now since it got removed from the conditions above